repotracker/repotracker.go

```diff
@@ -61,12 +61,40 @@ type VersionMetadata struct {
 	PeriodicBuildID     string
 }
 
+type ProjectInfo struct {
+	Ref                 *model.ProjectRef
+	Project             *model.Project
+	IntermediateProject *model.ParserProject
+}
+
+func (p *ProjectInfo) notPopulated() bool {
+	return p.Ref == nil || p.Project == nil
+}
+
+// PopulateVersion updates the version's ParserProject if we have or can create it
+func (p *ProjectInfo) populateVersion(v *model.Version) error {
+	if p.IntermediateProject != nil {
+		v.ParserProject = p.IntermediateProject
+		return nil
+	}
+	config, err := yaml.Marshal(p.Project)
+	if err != nil {
+		return errors.Wrap(err, "error marshaling config")
+	}
+	pp, err := model.LoadProjectInto(config, p.Ref.Identifier, p.Project)
+	if err != nil {
+		grip.Warning(message.WrapError(err, "error creating parser project"))
```

Comment:
I would have thought this should return an error instead of just logging a warning.